src/ORM/Table.php

```diff
@@ -460,17 +460,17 @@ public function schema($schema = null)
      * ### Example:
      *
      * ```
-     * protected function _initializeSchema(\Cake\Database\Schema\Table $table) {
+     * protected function _initializeSchema(\Cake\Database\Schema\TableSchema $table) {
      *  $table->columnType('preferences', 'json');
      *  return $table;
      * }
      * ```
      *
-     * @param \Cake\Database\Schema\Table $table The table definition fetched from database.
-     * @return \Cake\Database\Schema\Table the altered schema
+     * @param \Cake\Database\Schema\TableSchema $table The table definition fetched from database.
+     * @return \Cake\Database\Schema\TableSchema the altered schema
      * @api
      */
-    protected function _initializeSchema(Schema $table)
+    protected function _initializeSchema(TableSchema $table)
```

Comment:
I think #8612 is safe as no type hints changed. #8709 could cause issues, but I suspect fewer people have typehints against the HTTP/Client internals than override _initializeSchema()